src/test/operators/validate_test.cpp

```diff
@@ -85,6 +101,91 @@ TEST_F(OperatorsValidateTest, ScanValidate) {
   EXPECT_TABLE_EQ_UNORDERED(validate->get_output(), expected_result);
 }
 
+TEST_F(OperatorsValidateTest, ValidateAfterDelete) {
+  auto t1_context = Hyrise::get().transaction_manager.new_transaction_context();
+
+  auto validate1 = std::make_shared<Validate>(_gt);
+  validate1->set_transaction_context(t1_context);
+
+  validate1->execute();
+
+  EXPECT_EQ(validate1->get_output()->row_count(), 8);
+  t1_context->commit();
+
+  auto t2_context = Hyrise::get().transaction_manager.new_transaction_context();
+
+  // Select one row for deletion
+  auto table_scan = create_table_scan(_gt, ColumnID{0}, PredicateCondition::Equals, "13");
+  table_scan->execute();
+
+  auto delete_op = std::make_shared<Delete>(table_scan);
+  delete_op->set_transaction_context(t2_context);
+  delete_op->execute();
+
+  auto validate2 = std::make_shared<Validate>(_gt);
+  validate2->set_transaction_context(t2_context);
+  validate2->execute();
+
+  EXPECT_EQ(validate2->get_output()->row_count(), 7);
+  t2_context->commit();
+}
+
+TEST_F(OperatorsValidateTest, ChunkEntirelyVisibleThrowsOnRefChunk) {
+  if (!HYRISE_DEBUG) GTEST_SKIP();
+
+  auto snapshot_cid = CommitID{1};
+  auto pos_list = std::make_shared<PosList>();
+  Segments empty_segment = {std::make_shared<ReferenceSegment>(_test_table, ColumnID{0}, pos_list)};
+  auto chunk = std::make_shared<Chunk>(empty_segment);
+
+  EXPECT_THROW(Validate::is_entire_chunk_visible(chunk, snapshot_cid), std::logic_error);
+}
+
+TEST_F(OperatorsValidateTest, ChunkNotEntirelyVisibleWithoutMaxBeginCid) {
+  auto snapshot_cid = CommitID{1};
+  Segments empty_segment = {std::make_shared<ValueSegment<int32_t>>()};
+  auto chunk = std::make_shared<Chunk>(empty_segment, std::make_shared<MvccData>(0, 0));
```

Comment:
I'd feel more comfortable if we would not test on empty segments (other tests as well).

Reply:
Fair. For the test above using an empty segment, it should not make a difference since it throws because of the reference segment.

Reply:
> it should not make a difference since it throws because of the reference segment

Sure, but when testing edge cases, we should have one edge case at a time.